### PR TITLE
fix/15625: Workspace name is not present in header on connect bank account page

### PR DESCRIPTION
**Pull request by tienifr**
### Details

This PR will add the workspace name in the subtitle of the Connect bank account pages
### Fixed Issues

$ https://github.com/Expensify/App/issues/15625
PROPOSAL: https://github.com/Expensify/App/issues/15625#issuecomment-1463652473

### Tests

1. Open the app
2. Open any workspace
3. Open connect bank account
4. Verify that the workspace name is displayed on manual step page, start over page, error page, enable page (see the below screenshot for more detail)
- [x] Verify that no errors appear in the JS console

### Offline tests

Same above
### QA Steps

1. Open the app
2. Open any workspace
3. Open connect bank account
4. Verify that the workspace name is displayed on manual step page, start over page, error page, enable page (see the below screenshot for more detail)
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

1. Manual Step Page

Screen Shot 2023-03-19 at 13 49 33

2. Start Over Page
web15625-2

3. Error Page
web15625-4

4. Enable Page 

web15625-1

</details>

<details>
<summary>Mobile Web - Chrome</summary>

1. Manual Step Page

chrome15625-2

2. Start Over Page
chrome15625-3

3. Error Page

chrome15625-4

4. Enable Page 
chrome15625-1

</details>

<details>
<summary>Mobile Web - Safari</summary>

1. Manual Step Page
Screen Shot 2023-03-19 at 13 41 49

2. Start Over Page

Screen Shot 2023-03-19 at 13 40 38

3. Error Page

Screen Shot 2023-03-19 at 13 47 33

4. Enable Page 
Screen Shot 2023-03-19 at 13 47 10

</details>

<details>
<summary>Desktop</summary>

1. Manual Step Page

Screen Shot 2023-03-19 at 13 49 33

2. Start Over Page

web15625-2

3. Error Page
Screen Shot 2023-03-19 at 13 48 38

4. Enable Page 
web15625-3

</details>

<details>
<summary>iOS</summary>

1. Manual Step Page

Screen Shot 2023-03-19 at 14 06 13

2. Start Over Page

Screen Shot 2023-03-19 at 14 07 34

3. Error Page
Screen Shot 2023-03-19 at 14 08 27

4. Enable Page 
Screen Shot 2023-03-19 at 14 08 14

</details>

<details>
<summary>Android</summary>

1. Manual Step Page
Screen Shot 2023-03-19 at 14 22 43

2. Start Over Page

chrome15625-3

3. Error Page

Screen Shot 2023-03-19 at 14 21 54

4. Enable Page 
Screen Shot 2023-03-19 at 14 22 19

</details>

### CODE REVIEW

File: src/pages/ReimbursementAccount/BankAccountStep.js [hunk 1]
```diff
@@ -44,6 +44,9 @@ const propTypes = {
         /** Is the user account validated? */
         validated: PropTypes.bool,
     }),
+
+    /* The workspace name */
+    policyName: PropTypes.string.isRequired,
```
Comment (mollfpr) [inline src/pages/ReimbursementAccount/BankAccountStep.js:49]: I don't think we need `policyName` to be required. The page still accessible without providing `policyID` to the route params.

File: src/pages/ReimbursementAccount/ContinueBankAccountSetup.js [hunk 2]
```diff
@@ -37,6 +36,9 @@ const propTypes = {
         name: PropTypes.string,
     }).isRequired,
 
+    /* The workspace name */
+    policyName: PropTypes.string.isRequired,
```
Comment (mollfpr) [inline src/pages/ReimbursementAccount/ContinueBankAccountSetup.js:40]: Same as the above.

File: src/pages/ReimbursementAccount/EnableStep.js [hunk 3]
```diff
@@ -31,6 +32,9 @@ const propTypes = {
     /* Onyx Props */
     user: userPropTypes.isRequired,
 
+    /* The workspace name */
+    policyName: PropTypes.string.isRequired,
```
Comment (mollfpr) [inline src/pages/ReimbursementAccount/EnableStep.js:36]: Same as the above.

**Review by mollfpr — CHANGES_REQUESTED** [review]
Nicely done on this PR @tienifr.

Just a couple of comments on the prop types.

File: src/pages/ReimbursementAccount/BankAccountStep.js (hunk 1 above)
Comment (tienifr) [reply src/pages/ReimbursementAccount/BankAccountStep.js]: @mollfpr I just removed required and add the default policyName in all places. Could you help to review again

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



| Bank Account New | Bank Account Start Over | Bank Account Enable | Bank Account Error |
|------|------|----|----|
|16102 Web - New|16102 Web - Start over|16102 Web - Enable|16102 Web - Error|






</details>

<details>
<summary>Mobile Web - Chrome</summary>



| Bank Account New | Bank Account Start Over | Bank Account Enable | Bank Account Error |
|------|------|----|----|
|16102 mWeb:Chrome - New|16102 mWeb:Chrome - Start over|16102 mWeb:Chrome - Enable|16102 mWeb:Chrome - Error|






</details>

<details>
<summary>Mobile Web - Safari</summary>



| Bank Account New | Bank Account Start Over | Bank Account Enable | Bank Account Error |
|------|------|----|----|
|16102 mWeb:Safari - New|16102 mWeb:Safari - Start over|16102 mWeb:Safari - Enable|16102 mWeb:Safari - Error|







</details>

<details>
<summary>Desktop</summary>



| Bank Account New | Bank Account Start Over | Bank Account Enable | Bank Account Error |
|------|------|----|----|
|16102 Desktop - New|16102 Desktop - Start over|16102 Desktop - Enable|16102 Desktop - Error|






</details>

<details>
<summary>iOS</summary>



| Bank Account New | Bank Account Start Over | Bank Account Enable | Bank Account Error |
|------|------|----|----|
|16102 iOS - New|16102 iOS - Start over|16102 iOS - Enable|16102 iOS - Error|






</details>

<details>
<summary>Android</summary>



| Bank Account New | Bank Account Start Over | Bank Account Enable | Bank Account Error |
|------|------|----|----|
|16102 Android - New|16102 Android - Start over|16102 Android - Enable|16102 Android - Error|






</details>

**Review by mollfpr — COMMENTED** [review]
Well done @tienifr and thanks for raising the PR quickly 👍


@mountiny Tests well and all yours.

**Review by mollfpr — APPROVED** [review]
Well done @tienifr and thanks for raising the PR quickly 👍

@mountiny Tests well and all yours.

**Comment by tienifr** [comment]
@mountiny Can you take a look at this PR and provide feedback?

**Review by mountiny — APPROVED** [review]
Thank you @tienifr and @mollfpr

**Comment by kavimuru** [comment]
@tienifr How to trigger that error page? 

226159844-dea7f679-0a0e-4856-a667-2567f0876b2d

**Comment by mountiny** [comment]
@tienifr can you help with that?

@kavimuru even if you cant access it, I believe its same to assume this works for the error page if it works for the other pages too

**Comment by kavimuru** [comment]
@mountiny Yes it works in other pages.

**Comment by tienifr** [comment]
@kavimuru Sorry for my lateness. I see that there are 2 ways to access the error page
1. The field session.email has a value that ends with @expensify.sms 
Screen Shot 2023-03-27 at 11 16 02

2. Field throttedDate in the reimbursementAccount field in ONYX has value

Screen Shot 2023-03-27 at 11 14 14

cc @mountiny @mollfpr

**Pull request merged**